pkg/utils/kubernetes/crd.go

```diff
@@ -0,0 +1,43 @@
+// SPDX-FileCopyrightText: SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package kubernetes
+
+import (
+	"context"
+	"time"
+
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/gardener/gardener/pkg/client/kubernetes"
+	"github.com/gardener/gardener/pkg/utils/kubernetes/health"
+	"github.com/gardener/gardener/pkg/utils/retry"
+)
+
+// WaitUntilCRDManifestsReady takes manifests as strings and waits for them to get ready with a timeout of 15 seconds
+func WaitUntilCRDManifestsReady(ctx context.Context, c client.Client, manifests []string) error {
```

Comment:
When we spoke, didn't you plan on working with CRD names here instead, so unmarshalling is not required again?

Also, unit tests are missing for this function.